File: 3174. Clear Digits/3174. Clear Digits.py
class Solution:
    def clearDigits(self, s: str) -> str:
        res=[]

        for char in s:
            
            print("char",char," res",res)
            if (ord(char)>122 or ord(char)<97) and res:
                res.pop()
                continue
            res.append(char)
                
                
        return ''.join(res)

sol = Solution()
print(sol.clearDigits(s="fdd14"))
print(sol.clearDigits(s= "abc"))
print(sol.clearDigits(s = "cb34"))

File: 3174. Clear Digits/3174. Clear Digits.js
/**
 * @param {string} s
 * @return {string}
 */
var clearDigits = function(s) {
    let res = []
    for(let char of s){
        if ((char>'z' || char<'a') && res){
            res.pop()
            continue
        }
        res.push(char)
    }
    return res.join('')
};

console.log(clearDigits(s="abc"))
console.log(clearDigits(s="cb34"))